### PR TITLE
Create configuration to automatically generate release notes

### PR DESCRIPTION
**Pull request by itsjustkevin**
Github provides an API for [automatically generating release notes](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes). 

The structure of these release notes can be configured with a yml file. 

This PR is experimental and a step toward creating better release notes. 

This is a work in progress and subject to change as we understand how we want to organize the information provided.

### CODE REVIEW

**Comment by christopherfujino** [comment]
So if I'm reading https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes right, merging this will NOT generate any releases, but allow an option when manually creating a GitHub release to generate the notes?

**Comment by itsjustkevin** [comment]
@christopherfujino no releases will be generated.  This will allow us to utilize the API to generate release notes based on the parameters specified in the config file.

**Review by christopherfujino — APPROVED** [review]
LGTM

File: .github/release.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+changelog:
+  exclude:
+    authors:
+      - engine-flutter-autoroll
+      - fluttergithubbot
+    labels:
+      - passed first triage
```
Comment (devoncarew) [inline .github/release.yml:7]: This will exclude PRs that have the labels `passed first triage` / `passed secondary triage`?

**Comment by devoncarew** [comment]
> When using the rest API, this config file can be specified as a parameter to generate notes.

Ah, cool - I didn't see there was a REST API - I thought this file just controlled the text for when you create a release (having an API definitely makes this github provided functionality more useful).

We can always consider writing our own tool here, to gather PRs or commits since the last release, and emit a markdown file for our changelog.

File: .github/release.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  categories:
+    - title: Framework
+      labels:
+        - framework
+    - title: Tooling
+      labels:
+        - tool
+    - title: MacOS
+      labels:
+        - platform-mac
```
Comment (jmagman) [inline .github/release.yml:18]: Did you figure out if these are these processed in order? What happens to PR with `tool`, `platform-mac`, `passed first triage` and `passed secondary triage`?

File: .github/release.yml (hunk 1 above)
Comment (jmagman) [reply .github/release.yml:7]: I don't think any PRs would have these labels, only issues.

File: .github/release.yml (hunk 2 above)
Comment (itsjustkevin) [reply .github/release.yml:18]: I am still working this issue, It isn't reading from the release configs the way I want yet.  Will update.

File: .github/release.yml (hunk 1 above)
Comment (itsjustkevin) [reply .github/release.yml:7]: @devoncarew  @jmagman correct, this is still experimentation.

**Pull request merged**